components/ratgdo/ratgdo.cpp

```diff
@@ -724,6 +741,29 @@ namespace ratgdo {
         return *this->light_state;
     }
 
+    // Learn functions
+    void RATGDOComponent::activate_learn()
+    {
+        // Send LEARN with nibble = 0 then nibble = 1 to mimic wall control learn button
+        learn_poll_status_ = true;
+        this->send_command(Command::LEARN, 0);
+        set_timeout(150, [=] { this->send_command(Command::LEARN, 1); });
+        set_timeout(500, [=] { this->send_command(Command::GET_STATUS); });
+    }
+
+    void RATGDOComponent::inactivate_learn()
+    {
+        // Send LEARN twice with nibble = 0 to inactivate learn and get status to update switch state
+        this->send_command(Command::LEARN, 0);
+        set_timeout(150, [=] { this->send_command(Command::LEARN, 0); });
+        set_timeout(500, [=] { this->send_command(Command::GET_STATUS); });
+    }
+
+    void RATGDOComponent::toggle_learn()
+    {
+        this->learn_state = learn_state_toggle(*this->learn_state);
```

Comment:
Does this end up sending anything on the wire? From a quick look at the diff it seems to me that it doesn't?

Reply:
Doesn't seem to get called anywhere either so looks extraneous from the cut and paste go-by.